### DISCUSSION TITLE
[Question] How to compose functions depending of results of several functions?

### DISCUSSION THREAD

**Discussion by RabugenTom** [pointfreeco/swift-prelude#164] [Q&A]
This is not an issue with Prelude (or Overture) itself. I'm wondering if the following scenario is achievable with the operators shipping in both packages: given the functions
```swift
func assemble(_ x: Int, _ y: Int) -> String 
func combine(_ a: String, _ b: String) -> Bool
```
is it possible to recreate something equivalent to
```swift
var composed: (Int, Int, Int, Int) -> Bool = { x1, y1, x2, y2 in
  combine(assemble(x1, y1), assemble(x2, y2))
}
```
with operators or function composition? The all functions can be some curried variant of course, but I'm struggling to assemble two `assemble` functions next to each other without evaluating them first.
In other words, I'm trying to construct the following pipeline without prior knowledge of `x1, y1, x2, y2`:
```
      ┌──────────┐
x1 ->─┤x         │
      │ assemble ├──┐
y1 ->─┤y         │  │     ┌─────────┐
      └──────────┘  └──>──┤a        │
                          │ combine ├─>- Result
      ┌──────────┐  ┌──>──┤b        │
x2 ->─┤x         │  │     └─────────┘
      │ assemble ├──┘
y2 ->─┤y         │
      └──────────┘
```
Is such thing possible with Prelude or Overture?
Thanks!

**Comment by stephencelis** [comment]
@RabugenTom Going to convert this to a discussion :) I think it's certainly possible, though I'm not sure I'll have time to work on the exercise, but maybe someone else will!
